src/main/java/it/gov/pagopa/payment/service/qrcode/QRCodeCreationServiceImpl.java

```diff
@@ -75,6 +75,14 @@ public TransactionResponse createTransaction(
                 "Cannot find initiative with ID: [%s]".formatted(trxCreationRequest.getInitiativeId()));
       }
 
+      if (trxCreationRequest.getAmountCents().compareTo(0L) == 0) {
```

Comment:
when comparing Long you could just use equals or == if using native long instead of class Long

Reply:
maybe, we could check that it should be > 0